lib/bitcoin/wallet/utxo_handler.rb

```diff
@@ -0,0 +1,98 @@
+module Bitcoin
+  module Wallet
+    class UtxoHandler
+      attr_reader :watchings, :spv, :utxo_db, :pending_blocks, :pending_txs
+
+      # @param spv [Bitcoin::Node::SPV]
+      # @utxo_db [Bitcoin::Store::UtxoDb]
+      def initialize(spv, utxo_db)
+        @spv = spv
+        @spv.add_observer(self)
+        @utxo_db = utxo_db
+        @logger = Bitcoin::Logger.create(:debug)
+        @pending_blocks = []
+        @pending_txs = []
+      end
+
+      def update(event, data)
+        @logger.debug "UtxoHandler#update: #{event}, #{data}"
+        send(event, data)
+      end
+
+      private
+
+      # Called when receiving `tx` message from the connecting peers.
+      #
+      # - Store utxo if the output of received tx is in the watch list
+      # - Delete from utxo_db for spent output.
+      def tx(data)
+        tx = data.tx
+        tx.outputs.each_with_index do |output, index|
```

Comment:
Don't you need to check if it is a Tx that exists in the `merkleblock` that you received earlier?